NitroxClient/GameLogic/MobileVehicleBay.cs

```diff
@@ -20,20 +26,27 @@ public MobileVehicleBay(IPacketSender packetSender, Vehicles vehicles)
 
         public void BeginCrafting(ConstructorInput constructor, GameObject constructedObject, TechType techType, float duration)
         {
-            NitroxId constructorId = NitroxEntity.GetId(constructor.gameObject);
-
-            Log.Debug($"Building item from constructor with id: {constructorId}");
+            if (!TransmitLocalSpawns)
+            {
+                return;
+            }
 
-            VehicleModel vehicleModel = vehicles.BuildVehicleModelFrom(constructedObject, techType);
-            vehicles.AddVehicle(vehicleModel);
+            // Sometimes build templates, such as the cyclops, are already tagged with IDs.  Remove any that exist to retag.
+            UnityEngine.Component.DestroyImmediate(constructedObject.GetComponent<NitroxEntity>());
 
-            packetSender.Send(new ConstructorBeginCrafting(vehicleModel, constructorId, duration));
+            NitroxId constructedObjectId = NitroxEntity.GetId(constructedObject);
+            NitroxId constructorId = NitroxEntity.GetId(constructor.constructor.gameObject);
+            
+            VehicleWorldEntity vehicleEntity = new VehicleWorldEntity(constructorId, DayNightCycle.main.timePassedAsFloat, constructedObject.transform.ToDto(), "", false, constructedObjectId, techType.ToDto(), null);
+            vehicleEntity.ChildEntities = VehicleChildEntityHelper.ExtractChildren(constructedObject);
+                
+            packetSender.Send(new EntitySpawnedByClient(vehicleEntity));
 
             MonoBehaviour monoBehaviour = constructor.GetComponent<MonoBehaviour>();
             //We want to store the fallen position of the object to avoid flying object on reload 
             if (monoBehaviour)
             {
-                monoBehaviour.StartCoroutine(vehicles.UpdateVehiclePositionAfterSpawn(vehicleModel, constructedObject, duration + 10.0f));
+                monoBehaviour.StartCoroutine(vehicles.UpdateVehiclePositionAfterSpawn(constructedObjectId, techType, constructedObject, duration + 10.0f));
```

Comment:
Idk if you have plans for this next, but maybe this 10.0f should be replaced by ``VehicleWorldEntitySpawner.CONSTRUCTION_DURATION_IN_SECONDS`` if possible.